src/main/java/de/zalando/aruha/nakadi/validation/EventBodyMustRespectSchema.java

```diff
@@ -169,7 +165,17 @@ public JSONSchemaValidator(final JSONObject effectiveSchema) {
 
             return Optional.empty();
         } catch (final ValidationException e) {
-            return Optional.of(new ValidationError(e.getMessage()));
+            StringBuilder builder = new StringBuilder();
+            builder.append(e.getMessage());
+
+            if (!e.getCausingExceptions().isEmpty()) {
```

Comment:
I'm wondering if I should have a recursive implementation for errors on deeply nested structures. I'll try and tell.